src/GTP.cpp

```diff
@@ -547,7 +558,9 @@ bool GTP::execute(GameState & game, std::string xinput) {
             auto stonestring = game.board.get_stone_list();
             gtp_printf(id, "%s", stonestring.c_str());
         } else {
-            gtp_fail_printf(id, "Not a valid number of handicap stones");
+            gtp_fail_printf(id, 
+                "%d is not a valid number of handicap stones.\n"
+                "Use a number that is in the range 2 - 9", stones);
```

Comment:
Free handicap allows more than 9 stones. Also if you get into this condition then the option was missing, so "stones" won't have a valid value.